lib/Sema/TypeCheckAttr.cpp

```diff
@@ -6625,6 +6627,16 @@ void AttributeChecker::visitSendableAttr(SendableAttr *attr) {
         .warnUntilSwiftVersion(6);
     }
   }
+  // Prevent Sendable Attr from being added to methods of non-sendable types
+  if (auto *funcDecl = dyn_cast<AbstractFunctionDecl>(D)) {
+    if (auto baseType = funcDecl->getImplicitSelfDecl()->getTypeInContext()) {
+      if (!isSendableType(dc->getParentModule(), baseType)) {
```

Comment:
How does this behave for conditional conformances? e.g.

```swift
struct S<T> {
  let t: T

  @Sendable func test() {}
}

extension S: Sendable where T: Sendable {
  @Sendable func test2() {}
}
```

It seems like the first `@Sendable` should be diagnosed but the second should not.

Reply:
It should work out. `isSendableType()` calls `TypeChecker::conformsToProtocol()` which checks conditional requirements, and `getSelfTypeInContext()` maps the interface type of `self` into the function's own generic environment, so the generic parameters become archetypes that satisfy those conditional requirements. Worth adding a test though!